client/sites-dashboard/components/wpcom-smp-commands.tsx

```diff
@@ -89,6 +98,17 @@ export const useCommandsArrayWpcom = ( {
 		displaySuccessNotice( __( 'Copied new password' ) );
 	};
 
+	const openPHPmyAdmin = async ( siteId: number ) => {
```

Comment:
As this code is already in `phpadmin-card/index.js`, I believe that it would be a good idea to reuse that code.
What we can do is the following:
- Update `phpadmin-card/index.js` by spitting the following function:

```javascript
	async function openPmaLink() {
		setLoading( true );
		try {
			const { token } = await wpcom.req.post( {
				path: `/sites/${ siteId }/hosting/pma/token`,
				apiNamespace: 'wpcom/v2',
			} );

			if ( token ) {
				dispatch( trackOpenPhpmyadmin() );
				window.open( `https://wordpress.com/pma-login?token=${ token }` );
			}
		} catch {
			dispatch( errorNotice( translate( 'Could not open phpMyAdmin. Please try again.' ) ) );
		}
		setLoading( false );
	}
```

like this:

```javascript

export async function openPHPMyAdmin() {
			const { token } = await wpcom.req.post( {
				path: `/sites/${ siteId }/hosting/pma/token`,
				apiNamespace: 'wpcom/v2',
			} );

			if ( token ) {
				window.open( `https://wordpress.com/pma-login?token=${ token }` );
			}
	}
```

Then we can use it like so:
```javascript
const openPMALink = async () => {
 setLoading(true)
try{
 await openPHPMyAdmin()
dispatch( trackOpenPhpmyadmin() );
}catch(e){
  dispatch( errorNotice( translate( 'Could not open phpMyAdmin. Please try again.' ) ) )
}
setLoading(false)
}
```
- Reuse the extracted function in the `client/sites-dashboard/components/wpcom-smp-commands.tsx` file.

What do you think?

Reply:
I'm ok with reusing the code.
We can create a hook that returns this function:

```
function useOpenPmaLink() {
	const [ loading, setLoading ] = useState( false );
	const translate = useTranslate();
	const dispatch = useDispatch();

	const openPmaLink = useCallback(
		async function openPmaLink( siteId ) {
			setLoading( true );
			try {
				const { token } = await wpcom.req.post( {
					path: `/sites/${ siteId }/hosting/pma/token`,
					apiNamespace: 'wpcom/v2',
				} );

				if ( token ) {
					dispatch( trackOpenPhpmyadmin() );
					window.open( `https://wordpress.com/pma-login?token=${ token }` );
				}
			} catch {
				dispatch( errorNotice( translate( 'Could not open phpMyAdmin. Please try again.' ) ) );
			}
			setLoading( false );
		},
		[ dispatch, translate ]
	);

	return {
		openPmaLink,
		loading,
	};
}
```

This way we can handle the error by displaying the notice in both places.

This could be a follow-up PR.

Reply:
Hi @sejas @kozer  I added an error notice for now for cases when phpMyAdmin could not be opened. 

Since we are a bit on a timeline today, I think we could proceed with the following: I think we should merge as-is and handle the refactor in the follow-up PR. The main reason for that is that you are proposing two different, although similar solutions, and I would like to try both approaches and experiment with them without time pressure. How about creating a follow up issue and I can handle it next week once we are done with the high priority issues?

Reply:
Added the new issue here: https://github.com/Automattic/dotcom-forge/issues/4612